src/foam/box/socket/SocketConnectionBox.js

```diff
@@ -318,6 +324,26 @@ NOTE: duplicated in SocketConnectionReplyBox
         ((SocketConnectionBoxManager) getX().get("socketConnectionBoxManager")).remove(this);
       }
       `
+    },
+    {
+      name: 'releaseHoldingThread',
+      args: "Throwable t",
+      synchronized: true,
+      javaCode:`
+        int i = 0;
+        for (Map.Entry<String, BoxHolder> entry : ((Map<String, BoxHolder>) getReplyBoxes()).entrySet()) {
+          BoxHolder holder = entry.getValue();
+          Box replyBox = holder.getBox();
+          if ( replyBox != null ) {
+            Message reply = new Message();
+            reply.getAttributes().put("replyBox", replyBox);
+            reply.replyWithException(t);
+          }
+          getReplyBoxes().remove(entry.getKey());
+          i++;
+        }
+        getLogger().warning("SocketConnectionBox", getKey(), "free thread: " + i);
```

Comment:
```suggestion
        getLogger().warning("Terminating reply boxes", getKey());
```
The 'i' may be useful for debugging, but not in production.  And the logger already has the local class name along with host and port.